### PR TITLE
[ADD][9.0] framework_agreement

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
Ported module to v9

### CODE REVIEW

File: framework_agreement/__openerp__.py [hunk 1]
```diff
@@ -1,29 +1,16 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
 {'name': 'Framework Agreement',
  'summary': 'Long Term Agreement (or Framework Agreement) for purchases',
- 'version': '8.0.2.0.0',
+ 'version': '9.0.2.0.0',
  'author': "Camptocamp,Odoo Community Association (OCA)",
  'maintainer': 'Camptocamp',
  'category': 'Purchase Management',
  'complexity': 'normal',
```
Comment (elicoidal) [inline framework_agreement/__openerp__.py:12]: Remove this key (obsolete)

File: framework_agreement/data.xml [hunk 2]
```diff
@@ -1,10 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <openerp>
   <data>
-    <record id="framework_agreement_sequence_type" model="ir.sequence.type">
-      <field name="name">Framework Agreement</field>
-      <field name="code">framework.agreement</field>
-    </record>
     <record id="seq_mrp_repair" model="ir.sequence">
```
Comment (elicoidal) [inline framework_agreement/data.xml:4]: the id looks weird

File: framework_agreement/model/framework_agreement.py [hunk 3]
```diff
@@ -23,11 +11,10 @@
 from openerp.tools import DEFAULT_SERVER_DATE_FORMAT
 import openerp.addons.decimal_precision as dp
 
-AGR_PO_STATE = ('confirmed', 'approved',
-                'done', 'except_picking', 'except_invoice')
+AGR_PO_STATE = ('purchase', 'done', )
```
Comment (elicoidal) [inline framework_agreement/model/framework_agreement.py:14]: remove comma

File: framework_agreement/model/purchase.py [hunk 4]
```diff
@@ -190,14 +134,27 @@ def _propagate_fields(self):
...
         if self.framework_agreement_id:
             agreement = self.framework_agreement_id
             if agreement.supplier_id != self.order_id.partner_id:
-                raise exceptions.Warning(
+                raise UserError(
                     _('Invalid agreement '
-                      'Agreement and supplier does not match')
-                )
-
+                      'Agreement and supplier does not match'))
```
Comment (elicoidal) [inline framework_agreement/model/purchase.py:142]: do not match

File: framework_agreement/model/purchase.py [hunk 5]
```diff
@@ -190,14 +134,27 @@ def _propagate_fields(self):
...
+        for record in self:
+            if record.framework_agreement_id:
+                agreement_price = record.framework_agreement_id.get_price(
+                    record.product_qty,
+                    currency=record.currency_id)
+                if agreement_price != self.price_unit:
+                    msg = _(
+                        "In line %s You have set the price to %s \n"
+                        " but there is a running agreement"
+                        " with price %s") % (
```
Comment (elicoidal) [inline framework_agreement/model/purchase.py:158]: Is it possible that the PO is in USD and agreement in EUR and price been the same would pass the validation?

Comment (elicoidal) [inline framework_agreement/model/purchase.py:158]: My point is I didnot see any check on the currency of the PO vs agreement

**Comment by JordiBForgeFlow** [comment]
After further analysis I'm seeing that there's an overlap between "framework.agreement" and "supplier.pricelist" in v9, in the sense that they both introduce prices for given dates and quantities, for a given currency.

So I'm not convinced at all that the framework agrements shoud be ported to v9, using the current proposed approach.

cc @pedrobaeza

**Pull request closed without merging**